### PR TITLE
Add rule-engine recipe

### PR DESCRIPTION
**Pull request by teake**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by teake** [comment]
@conda-forge/help-python, ready for review!

File: recipes/rule-engine/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.4
+  run:
+    - ply >=3.9
+    - python >=3.4
+    - python-dateutil ~=2.7
```
Comment (BastianZim) [inline recipes/rule-engine/meta.yaml:25]:
```suggestion
    - python-dateutil >=2.7,==2.*
```
Condo-build can't really understand `~=`(Only through some hacks) so it's better to use the conda-build equivalent.

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just one small thing, then this is ready to go 🚀

File: recipes/rule-engine/meta.yaml (hunk 1 above)
Comment (teake) [reply recipes/rule-engine/meta.yaml]: That didn't work. I've changed it to `>=2.7,<3` as in the [examples in the conda docs](https://conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications).

Btw, just so I understand, is it documented somewhere that conda-build cannot understand `~=` properly?

Comment (BastianZim) [reply recipes/rule-engine/meta.yaml]: Hmm strange. @marcelotrevisani this is what's suggested by grayskull but it didn't work. Do you know why?

> Btw, just so I understand, is it documented somewhere that conda-build cannot understand ~= properly?

Conda-build can only understand the following: https://docs.conda.io/projects/conda-build/en/latest/resources/package-spec.html#build-version-spec
We are patching some other things in the background (like the one here) but it's better to be on the safe side.

Comment (teake) [reply recipes/rule-engine/meta.yaml]: Ok, thanks for the clarification!

**Pull request merged**